src/nvim/ex_cmds.c

```diff
@@ -3723,32 +3718,53 @@ static buf_T *do_sub(exarg_T *eap, proftime_T timeout,
            */
           while (subflags.do_ask) {
             if (exmode_active) {
-              char_u      *resp;
-              colnr_T sc, ec;
+              if (ex_normal_busy) {
```

Comment:
It doesn't work by overriding the filled char below to be `'\n'` when in ex mode confirm mode? it would be good to avoid an extra loop for such a particular special case upfront.

https://github.com/neovim/neovim/blob/2e156a3b7d7e25e56b03683cc6228c531f4c91ef/src/nvim/getchar.c#L2242

Reply:
You're right. That should be just enough to mimic Vim behaviour.